probe-rs/src/config/flash_algorithm.rs

```diff
@@ -0,0 +1,34 @@
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct FlashAlgorithm {
+    /// Memory address where the flash algo instructions will be loaded to.
+    pub load_address: u32,
```

Comment:
If there's a fixed load address, why is the code position-independent? Is the load address just a suggestion?

Reply:
It is not required to put the flash there. Basically how it is done atm is that the load_address is calculated when parsing the algorithm from the pack. And lateron it is loaded into RAM at this address. The ELF binary itself is position independent.

Reply:
I fixed this by making the entire algorithm config completely memory indepent. The full, final algorithm is then assembled using the memory properties of the exact selected chip :)